cpu/avr8_common/include/cpu.h

```diff
@@ -146,14 +185,33 @@ static inline void avr8_enter_isr(void)
  */
 static inline int avr8_is_uart_tx_pending(void)
 {
-    uint8_t state = avr8_get_state();
-    return (state & (AVR8_STATE_FLAG_UART0_TX
-                  |  AVR8_STATE_FLAG_UART1_TX
-                  |  AVR8_STATE_FLAG_UART2_TX
-                  |  AVR8_STATE_FLAG_UART3_TX
-                  |  AVR8_STATE_FLAG_UART4_TX
-                  |  AVR8_STATE_FLAG_UART5_TX
-                  |  AVR8_STATE_FLAG_UART6_TX));
+    uint8_t state;
+
+    __asm__ volatile(
+        "lds   %[state], avr8_uart_state \n\t"
+        : [state]   "=r" (state)
+        :
+        : "memory"
+    );
+
+    return state != 0;
+}
+
+/**
+ * @brief        Set UART TX channel as pending
+ *
+ * @param uart   The UART number
+ * @param value  >0 set pending
+ *                0 unset pending
+ */
+static inline void avr8_uart_tx_set_pending(unsigned uart, unsigned value)
```

Comment:
IMO it is easier to readyl to have two functions, one for setting and one for clearing. Due to the `static inline`, both should yield the same machine code. So there should be no harm in it.

Reply:
done!